### DISCUSSION TITLE
How can I allow only metamask wallets when connecting wallets through walletconnect?

### DISCUSSION THREAD

**Discussion by siosio34** [WalletConnect/walletconnect-monorepo#2185] [Developer Support]
How can I allow only metamask wallets when connecting wallets through walletconnect?

**Comment by glitch-txs** [comment]
Do you mean by scanning the QR or on the modal?

**Reply by siosio34** [reply]
@glitch-txs 
you're right. However, at that time, if the type of wallet is unknown and cannot be prevented, we are considering a method of checking whether it is a Metamask wallet immediately after connection and displaying an error message.

**Reply by glitch-txs** [reply]
May I know why you want to allow only MM? Just curious. And yep I don't think there's a direct way of doing it if you mean by restricting the QR scan.

**Reply by siosio34** [reply]
The reason is that, from what I understand, it seems that other wallets other than Metamask are still a security threat within the company.

Once, I found a field that seemed to mean the wallet name from the connection object and processed it as follows.
I used WAGMI and I don't know if this is correct, but I attach the code below.

here is my code
```tsx

      const connectResult = await connect({
        connector: connectors.find(
          (connector) => connector.id === currentConnectorId
        ),
      })

      const isMetamask =
        connectResult.connector.id === 'metaMask' ||
        (connectResult.provider as any)?.provider?.signer?.connection?.wc
          ?._peerMeta.name === 'MetaMask'

      if (!isMetamask) {
        throw new NotMetaMaskConnectorError()
      }

```

**Reply by glitch-txs** [reply]
This was answered on [WalletConnect discord server](https://discord.com/channels/492410046307631105/1096367156884688958/1097083767396503573), just saw that. It's currently not supported but it's a feature they will implement on the next releases.
